tensorboard/data/server/logdir.rs

```diff
@@ -246,22 +263,27 @@ impl<'a> LogdirLoader<'a> {
             .runs
             .read()
             .expect("could not acquire runs.data");
+
+        let mut work_items = Vec::new();
         for (run, run_state) in self.runs.iter_mut() {
             let event_files = discoveries
                 .remove(run)
                 .unwrap_or_else(|| panic!("run in self.runs but not discovered: {:?}", run));
             let filenames: Vec<PathBuf> = event_files.into_iter().map(|d| d.event_file).collect();
-            run_state.loader.reload(
-                filenames,
-                commit_runs.get(run).unwrap_or_else(|| {
-                    panic!(
-                        "run in self.runs but not in commit.runs \
+            let run_data = commit_runs.get(run).unwrap_or_else(|| {
+                panic!(
+                    "run in self.runs but not in commit.runs \
                         (is another client mutating this commit?): {:?}",
-                        run
-                    )
-                }),
-            );
+                    run
+                )
+            });
+            work_items.push((&mut run_state.loader, filenames, run_data));
```

Comment:
Just to check my understanding, this is a `Vec` of the parameters to the closures we ultimately run, rather than the closures themselves, because the closures even though they are all identically structured here are in fact all different anonymous types, so we can't put them in a `Vec` directly (we'd have to used a boxed `FnMut` or something).

Reply:
Okay, got it, thanks.  I think my natural inclination would be to do a vector of argument-less closures since that seems more encapsulated (the work item execution code doesn't need to know anything task-specific about how to unpack the vec), but I don't have a strong preference.